services/storage/store.go

```diff
@@ -311,6 +355,16 @@ func (s *Store) TagKeys(ctx context.Context, req *datatypes.TagKeysRequest) (cur
 		if found := reads.HasFieldValueKey(expr); found {
 			return nil, errors.New("field values unsupported")
 		}
+		if found := reads.ExprHasKey(expr, fieldKey); found {
```

Comment:
This is going to be extremely expensive to compute, right? Previously we could always consult the series index for this call, but now we have to go to TSM for every series key/field key combination to find out if data exists for that key.

Do we expect the UI features that require this to be usable for large databases?